### ISSUE TITLE
Can't infer a dim denoted by -1 in function 'computeShapeByReshapeMask' [ONNX]

### ISSUE THREAD

**Issue by Palmitoxico** [opencv/opencv#16140]
##### System information (version)
- OpenCV => 4.2-pre (9a18330f3a1a5eddb5e4f23e8daa9534d83fa8c6)
- Pytorch => tried with 1.4 and 1.1, same results
- Operating System / Platform => ArchLinux x86_64
- Compiler => gcc

##### Detailed description

I'm trying to load and evaluate a dnn model trained with pytorch, but having no luck so far. OpenCV seems to load the onnx file just fine, but it throws an exception when calling net.forward(). I've tried with different models but I keep getting exceptions:

```resnet18```:
```
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.2.0-pre) /home/giachero/repos/opencv/modules/dnn/src/layers/reshape_layer.cpp:154: error: (-1:Backtrace) Can't infer a dim denoted by -1 in function 'computeShapeByReshapeMask'
```

```resnet34```:
```
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.2.0-pre) /home/giachero/repos/opencv/modules/dnn/src/layers/reshape_layer.cpp:154: error: (-1:Backtrace) Can't infer a dim denoted by -1 in function 'computeShapeByReshapeMask'
```

```squeezenet```:
```
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.2.0-pre) /home/giachero/repos/opencv/modules/dnn/src/layers/reshape_layer.cpp:113: error: (-215:Assertion failed) total(srcShape, srcRange.start, srcRange.end) == maskTotal in function 'computeShapeByReshapeMask'
```

##### Steps to reproduce

Generate an untrained ONNX model with pytorch:
```python
#!/usr/bin/env python3

from torch.autograd import Variable
import torch
import torch.onnx
import torchvision
import torch.nn as nn

model = torchvision.models.resnet18(num_classes=2)
# model = torchvision.models.resnet34(num_classes=2)
# model = torchvision.models.squeezenet(num_classes=2)

model = torch.nn.DataParallel(model)

model.eval()

model = model.module.cpu()

dummy_input = Variable(torch.randn(10, 3, 224, 224))

input_names = [ "data" ]
output_names = [ "fake" "genuine" ]

torch.onnx.export(model, dummy_input, "dummy.proto", verbose=False, input_names=input_names, output_names=output_names)
```

Load and execute the ONNX model:
```python
#!/usr/bin/env python3

import cv2 as cv
import sys

try:
    image = sys.argv[1]
    onnx_model = sys.argv[2]
except:
    print("Usage: " + sys.argv[0] + " image onnx_model")
    exit(1)

net = cv.dnn.readNetFromONNX(onnx_model)
net.setPreferableBackend(0)
net.setPreferableTarget(0)
image = cv.imread(image)
blob = cv.dnn.blobFromImage(image, size=(224, 224), swapRB=True, ddepth=cv.CV_8U)
net.setInput(blob)

print (net.forward())

```

Augusto.

**Comment by dkurt** [comment]
Cannot reproduce it with PyTorch 1.2.0

**Comment by Palmitoxico** [comment]
It seems that my pytorch installation was messed up (multiple versions installed). I have cleaned everything and built from source and it works now. Thanks!
